41-Notificar al usuario cuando un Rent se vence

paul-ot opened the pull request:
- Establecí como 'Buenos Aires' el time zone, para trabajar directame con GMT-3.
- Añadida gema sidekiq-scheduler.
- Añadido worker para agendar tareas con Sidekiq.
- Añadido template de email para notificar al dueño de un rent cuando este vence.
- Añadido método en RentMailer, para enviar el email de vencimiento de rent.
- Añadido trigger en Rent model, para crear el job de vencimiento que enviará el email.

paul-ot committed:
- Establecí como 'Buenos Aires' el time zone, para trabajar directamente con GMT-3.
- Añadida gema sidekiq-scheduler.
- Añadido worker para agendar tareas con Sidekiq.
- Añadido template de email para notificar al dueño de un rent cuando este vence.
- Añadido método en RentMailer, para enviar el email de vencimiento de rent.
- Añadido trigger en Rent model, para crear el job de vencimiento que enviará el email.

holywyvern commented on app/models/rent.rb:
```diff
@@ -7,11 +7,20 @@ class Rent < ApplicationRecord
+  def schedule_end_email
+    Sidekiq.set_schedule('rent_end_email_worker',
+                         {
+                             at: self.end_date.to_s + ' 09:00:00',
```
Creo que es mas claro si haces un `end_date + 9.hours` y hablando de eso, solemos obviar self a menos que sea necesario (como cuando haces `self.atributo = valor`)

paul-ot replied:
Fixed.

holywyvern commented on app/models/rent.rb:
```diff
@@ -7,11 +7,20 @@ class Rent < ApplicationRecord
+                         {
+                             at: self.end_date.to_s + ' 09:00:00',
+                             class: 'RentEndEmailWorker',
+                             args: {rent_id: self.id}
```
Lo mismo, no necesitas self.

paul-ot replied:
Fixed.

holywyvern commented on config/application.rb:
```diff
@@ -17,5 +17,6 @@ class Application < Rails::Application
     # the framework and any gems in your application.
 
     config.autoload_paths << Rails.root.join('app', 'poros')
+    config.time_zone = 'Buenos Aires'
```
Guarda con esto, deberias de setear el timezone de la fecha y no de la aplicación entera

paul-ot replied:
Fixed. Removed this config line, and now we're setting up the time zone per request, in the rents controller, create method.

paul-ot replied:
fixed

paul-ot committed:
- Removed config for timezone.
- Implemented `at: (end_date + 9.hours).in_time_zone("Buenos Aires")` in Sidekiq.set_schedule, to convert and schedule the datetime value in the GMT-3 datetime.
